[Customer Home]: Make Customer Home the default landing page for logged-in users

ramonjd opened the pull request:
## Changes proposed in this Pull Request

We removed the '_alternative default logged-in landing **p**age_' A/B test in #39475. 

#39475 tested redirecting logged-in users to the Customer Home rath**e**r than the Reader when l**a**nding on `/`.

This PR re-enables the default redire**c**t to Customer Home. 

There is no A/B t**e**st.

We're als**o** [checking if the **u**ser can use Customer Home](https://github.com/Automattic/wp-calypso/blob/master/client/state/sites/selectors/can-current-user-use-customer-home.js#L25) for the given primary blog ID.

The fallback redirec**t** remains `/read`.

## Testing instructions

For best results, queue up some [soothing testing music.](https://www.youtube.com/watch?v=su0mR4_z2so)

### Test the redirect

- Create a new site
- Make it your primary site over at `/me/account`
- Refresh the page at `/`
- You should land on Customer Home for your primary site
- Reflect on the good things, even the ephemeral

### Test the fallback

- Create an account, but not a site at `/start`
- Refresh the page at `/`
- You should land on the Reader
- Find new virtues in difficult exercises

The `canCurrentUserUseCustomerHome` selector returns `false` for [VIP sites, Jetpack sites and sites in the control group for the `customerHomeAll` A/B test](https://github.com/Automattic/wp-calypso/blob/master/client/state/sites/selectors/can-current-user-use-customer-home.js#L25).

You can either edit this selector to return `false`, or:

- Make a site created before `2019-08-06` your primary site
- Make sure the `customerHomeAll` A/B test is set to `control`
- You should land on the Reader
- Note that, though we are small, we can make big changes.

## Related reading
p8Eqe3-Vq-p2#comment-3232

ramonjd committed:
We removed the Reader A/B test in #39475. This PR re-enables the default redirect to customer home without the abtest variation split.
We're also checking if the user can use customer home for the given primary blog.
The fallback redirect remains `/read`.

roo2 commented:
There is an issue for sites with custom domains: If you have a primary site with a custom domain, e.g. `blagmusic.art.blog` and access `/` you will get redirected to the site selector and the url will contain the `*.wordpress.com` url for that site e.g. `home?site=blagmusicart.wordpress.com`. It looks like these sites need to use the actual domain name instead of the `primarySiteSlug`

ramonjd commented:
> There is an issue for sites with custom domains: If you have a primary site with a custom domain, e.g. TEST_SITENAME.blog and access / you will get redirected to the site selector and the url will contain the *.wordpress.com url for that site e.g. home?site=TEST_SITENAME.wordpress.com.

Ah, nice observation. I'll check it out 👍

ramonjd commented:
I've checked this against a bought domain, e.g., `.com` and against a newly-created blog (`TEST_SITENAME.travel.blog`)

The redirect works for both, but you're right, in the case of free blog domains, it's a little unsightly: 

`/home/TEST_SITENAME.travel.blog?site= TEST_SITENAME.wordpress.com`

We could grab the site's URL from `state.sites.items[SITE_ID].URL`, and snip off the `https://`. 

Maybe for the next iteration?

roo2 commented:
> The redirect works for both, but you're right, in the case of free blog domains, it's a little unsightly:
> `/home/TEST_SITENAME.travel.blog?site= TEST_SITENAME.wordpress.com`

hmm for me it hit a different page: there must have been something else going on for the sites I tested with beyond just having a domain name set up, for me, it just went to `/home?site=TEST_SITENAME.wordpress.com`:

<img width="337" alt="Screen Shot 2020-02-25 at 2 25 25 PM" src="https://user-images.githubusercontent.com/22446385/75216710-04569100-57e9-11ea-8ca4-5f7605b86513.png">


I'll try and work out if there's something else special about my test sites...

roo2 commented:
hmm yea, I signed up for a new site with the free domain `erutestredirectsite.design.blog` and then set it to my default site. when I set the route to `/`, I got this page:

<img width="706" alt="Screen Shot 2020-02-25 at 4 13 18 PM" src="https://user-images.githubusercontent.com/22446385/75217079-1b49b300-57ea-11ea-940a-6e12890a3486.png">

simison commented:
Just FYI @gwwar since your team is working on customer home and this change is likely affecting any metrics you follow. :-)

gwwar commented:
@m @amamujee would you prefer if we 100% Customer Home as being visible to everyone too? We're currently soft rolling the section being available to everyone. (Was limited before to sites created after Aug 2019). See also p8Eqe3-15Y-p2 for context.

In terms of ordering I'd prefer having this section fully available first, then updating the default landing page, since the behavior will bifurcate otherwise. Happy to just turn on the section for everyone if folks are comfortable.

ramonjd commented:
> In terms of ordering I'd prefer having this section fully available first, then updating the default landing page, since the behavior will bifurcate otherwise. Happy to just turn on the section for everyone if folks are comfortable.

Deferring to you on this one @gwwar Thanks for chiming in. 

We can put this on ice until you're ready, if that's the consensus. I'll keep it on my watch list.

tyxla reviewed (commented):
Seems like this needs a rebase now

ramonjd committed:
This commit takes into account the refactor of the logged-in and logged-out middleware in #39460 by moving the default logged-in route handlers to root.js.

We're preferencing state selectors over user() as we would like to eventually deprecate lib/user and replace it with the Redux store. See #39460#discussion_r380085303 for context.

ramonjd commented:
> Seems like this needs a rebase now

Thanks @tyxla 

I've rebased and refactored in light of the changes in #39460

cc @enejb @robertf4 who worked on the new entry point changes. 🙇

kwight commented:
@gwwar Feel like this is in limbo – do we need a decision from someone on opening Home to 100%, before we move forward with this?

gwwar commented:
Is it in limbo? I lost track of pings while at the meetup.

My comment was a question, it wasn't blocking. In terms of timing, we'll likely 100% Customer Home on Monday, so whatever folks prefer.

I'd recommend trying a bigger change like this on not Friday though, with some heads up to folks :)

robertf4 commented:
From the perspective of the changes in #39460, I think this looks good 👍

amamujee commented:
My instinct here is to accelerate the rollout for Customer Home to everyone and then make this the default landing page vs. the reader. We’ve not seen much impact on metrics or users reporting issues and it’ll incentivize us to further invest and fix Customer Home more incrementally for users. It’s also in the spirit of getting all our users on the same product experience.

gwwar commented:
I'll prepare a patch and will link back 👍

gwwar commented:
Created https://github.com/Automattic/wp-calypso/pull/39880 if folks didn't have objections we can likely land this tomorrow and see how it does.

ramonjd committed:
This commit uses the site slug of the primary site id in the redirect url, which returns any custom domain names, for example, *.blog rather than the .wordpress.com primary site slug.

ramonjd commented:
> hmm yea, I signed up for a new site with the free domain erutestredirectsite.design.blog and then set it to my default site. when I set the route to /, I got this page:

Good spotting @roo2 

9b63f00ca8679d6c0a5d60af41c26689f12781e0 should fix that 👍

andrewserong reviewed (approved):
LGTM! 🚢  Tested across:

* Jetpack site set as primary site (redirects to Reader as expected as it doesn't have customer home)
* Old site that's pre-customer home (redirects to Reader as expected)
* Newer site with `.wordpress.com` domain (redirects to Customer Home)
* Newer site with `.food.blog` domain (redirects to Customer Home)

ramonjd commented:
We're reviewing this @gwwar 

It should be ready to roll on Monday after #39880 is shipped.

There is one minor glitch worth mentioning. 

When performing a hard refresh on `/` with a clear cache, we think a race condition is occurring between: 

- the conditional checks we're using to redirect, and 
- the contents of our store.

That is, whatever we're getting from `/me` and `/sites` isn't being pushed into the state and cached **before** we redirect. 

Subsequent refreshes work.

The worse that will happen is that the user will get to enjoy the Reader once.

The consensus is that it's not a deal breaker.

roo2 commented:
> hmm for me it hit a different page: there must have been something else going on for the sites I tested with beyond just having a domain name set up, for me, it just went to `/home?site=TEST_SITENAME.wordpress.com`:


^^ That issue is fixed with the latest changes  👍
